imap_processing/swe/l1b/swe_l1b_science.py

```diff
@@ -254,21 +205,31 @@ def apply_in_flight_calibration(
     Parameters
     ----------
     corrected_counts : numpy.ndarray
-        Corrected count of quarter cycle data. Data shape is (180, 7).
-    acquisition_time : float
-        Acquisition time of current quarter cycle data.
+        Corrected count of full cycle data. Data shape is (24, 30, 7).
+    acquisition_time : numpy.ndarray
+        Acquisition time of full cycle data. Data shape is (24, 30, 7).
 
     Returns
     -------
     corrected_counts : numpy.ndarray
-        Corrected count of quarter cycle data after applying in-flight calibration.
-        Array shape is (180, 7).
+        Corrected count of full cycle data after applying in-flight calibration.
+        Array shape is (24, 30, 7).
     """
-    # calculate calibration factor
-    cal_factor = calculate_calibration_factor(acquisition_time)
-    # Apply to current quarter cycle data
+    # Read in in-flight calibration data
+    in_flight_cal_df = read_in_flight_cal_data()
+    # calculate calibration factor.
+    # return shape of interp_func is (24, 30, 7, 7) where last 7 dimension contains
+    # calibration factor for each CEM detector.
+    cal_factor = interp_func(
+        acquisition_time,
+        in_flight_cal_df["met_time"].values,
+        in_flight_cal_df.iloc[:, 1:].values,
+    )
+    # Apply to full cycle data
     corrected_counts = corrected_counts.astype(np.float64)
-    corrected_counts *= cal_factor
+    for idx in range(7):
+        cem_cal_factor = cal_factor[:, :, idx, idx]
+        corrected_counts[:, :, idx] *= cem_cal_factor
```

Comment:
This should be a completely broadcastable multiplication indicating you have one cal factor per value in the array.
```suggestion
    corrected_counts *= cem_cal_factor
```

You are also taking in `corrected_counts`, but I think that is the returned quantity? I'd separate out the input/output array names and avoid in-place modification, maybe calling the input out explicitly.
```python
return input_data * cem_cal_factor
```